components/ai_chat/resources/page/state/conversation_context.tsx

```diff
@@ -538,20 +540,24 @@ export function ConversationContextProvider(props: React.PropsWithChildren) {
             (getImageFiles(turn.uploadedFiles)?.length || 0),
           0
         )
-        const currentPendingImages = context.pendingMessageImages?.length || 0
+        const currentPendingImages = context.pendingMessageImages.length
         const maxNewImages = MAX_IMAGES - totalUploadedImages - currentPendingImages
         const newImages = images.slice(0, Math.max(0, maxNewImages))
 
         if (newImages.length > 0) {
           setPartialContext({
-            pendingMessageImages: context.pendingMessageImages
+            isUploadingFiles: false,
+            pendingMessageImages: context.pendingMessageImages.length > 0
               ? [...context.pendingMessageImages, ...newImages]
               : [...newImages]
```

Comment:
```suggestion
            pendingMessageImages: [...context.pendingMessageImages, ...newImages]
```

Reply:
Fixed in https://github.com/brave/brave-core/pull/29017/commits/3b150f67df84bf03877781a396161fd2cde60ce0